docs/guides/firmware-update.md

```diff
@@ -0,0 +1,98 @@
+the following:
+
+- SPI flash write-protection - prevents overwriting the initial bootblock and
+  Verified Boot code
+- SMM BIOS write protection - prevents all writes to BIOS flash memory outside
+  of privileged code running in System Management Mode
+- UEFI Secure Boot - in combination with Linux Kernel Lockdown, prevents direct
+  access to the SPI flash controller from the OS
+
+If you are interested, see the
+[Dasharo System Features](../dasharo-menu-docs/dasharo-system-features.md)
+article for more details.
+
+To allow updating firmware by the end user, these protections must be disabled
+first. To facilitate this, Dasharo has a Firmware Update Mode option that
+**temporarily** disables firmware security measures for the duration of one
+boot.
+
+To enter Firmware Update Mode:
+
+1. Enter the Setup Menu:
+![](./images/setup_menu_dsf.png)
+1. Navigate to `Dasharo System Features`:
+![](./images/setup_menu_dsc.png)
+1. Navigate to `Dasharo Security Options`:
+![](./images/setup_menu_lbm.png)
+1. Select the `Enter Firmware Update Mode` option:
+![](./images/setup_menu_fum.png)
+1. When prompted, press Enter to accept. The device will reboot in Firmware
+  Update Mode.
+1. When prompted, press the indicated key on the keyboard. Alternatively, to
+  abort Firmware Update Mode, press Enter instead or simply wait for the timeout
+  to expire.
+
+Once in Firmware Update Mode, proceed with the firmware update steps outlined
+in device-specific documentation.
+
+## Firmware Update Mode flowchart
+
+```mermaid
+flowchart TD
+    power_on[Power on] --> setup
+    subgraph setup [Setup UI]
+    setup1[Enter Dasharo Setup]
+    setup2[Go to Dasharo System Features]
+    setup3[Go to Dasharo Security Options]
+    setup4[Choose Firmware Update Mode]
+    setup5[Press ENTER to continue]
+    setup1 --> setup2 --> setup3 --> setup4 --> setup5
+    end
+    setup --> reboot[Reboot]
+    reboot --> coreboot
+    subgraph coreboot
+    cb1{{Firmware Update Mode EFI variable found?}}
+    cb1 -- False --> cb2[Enable firmware lockdown]
+    cb1 -- True --> cb3{{Firmware Update Mode EFI variable value?}}
+    cb3 -- False --> cb2
+    cb3 -- True --> cb4[Skip firmware lockdown]
+    cb2 --> cb5{{Boot payload}}
+    cb4 --> cb5
+    end
+
+    coreboot --> securitypkg
+    subgraph securitypkg [SecurityPkg]
+    sec1{{Firmware Update Mode EFI variable found?}}
+    sec1 -- False --> sec2[Enable Secure Boot]
+    sec1 -- True --> sec3[Skip enabling Secure Boot]
+    sec2 --> sec4[Continue booting]
+    sec3 --> sec4
+    end
+
+    securitypkg --> bootmanager
+    subgraph bootmanager [PlatformBootManagerLib]
+    bm1{{Firmware Update Mode EFI variable found?}}
+    bm1 -- False --> bm2[Continue booting]
+    bm1 -- True --> bm3[Remove Firmware Update Mode EFI variable]
+    bm3 --> bm4[Display Firmware Update Mode warning dialog]
+    bm4 --> bm5[Confirm user presence]
+    bm5 --> bm2
+    end
+
+    bootmanager --> userpresence{{User presence confirmed?}}
```

Comment:
if FUM variable is not found then we should indeed continue booting. However the way it is shown on diagram it looks like user presence confirmation is checked on both paths on FUM and non-FUM.

The way I would see it:

If FUM variable is not set or not found, then go to "Continue booting" -> kill (no more arrows)

If FUM variable is set:

Remove Firmware Update Mode EFI variable -> 
Display Firmware Update Mode warning dialog ->
Confirm user presence ->
User presence confirmed?
-> (True) Boot OS
-> (False) Reboot (maybe an arrow should go back to the first reboot right before coreboot)
kill (no more arrows)

I would not add anything more outside PlatformBootManagerLib block.

Reply:
Updated in 957d3c3d2b4e2c1f9d38e70eb8f6fabd4463c196

> Reboot (maybe an arrow should go back to the first reboot right before coreboot)

Can't do that with mermaid.js, without having the layout of the flowchart changing entirely (platformbootmanagerlib moves to the top then)

Reply:
@mkopec one arrow is still wrong. The true arrow from `Confirm user presence` should not start from `Confirm user presence` but from `User presence confirmed?`.  `Confirm user presence` is not a question, so no true/false arrow should start from there

Reply:
Switched the diagram to UML, please review again @miczyg1 @macpijan

Reply:
I see we did a redesign of the diagram again... How am I supposed to close this thread?

Now I see that on the non-FUM path in the PlatformBootManagerLib we still have `User presence check failed?` which should not be there... There should be an arrow to `Boot` directly from `Do nothing`.

Reply:
Also if we have Boot, why do we `Present boot menu` if FUm is not set? What boot menu? And why?

On non-FUM path it should be `Do nothing` -> `Boot` -> kill, that's it.

On FUM path we should have `Confirm User Presence` -> `User presence check failed?` -> (false) `Boot iPXE` -> `Boot DTS via iPXE` -> `Start Firmware Update`. Why is this so complicated?

Reply:
That's why I asked how much of the implementation details need to be here. Applied your suggestions in bfe41ce1931cbd57f2a1bb753bb84ae19e551328.

Reply:
Now looks good. The arrows from `Set boto entry to iPXE` to `Boot` look ugly, but I guess we can't do much about it, it's just plain UML....